File: user-service/src/main/java/rs/edu/raf/dto/EmployeeCreateDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;

import java.util.List;

public record EmployeeCreateDTO (
        Long companyId,
        @Pattern(regexp = "^[a-zA-Z]+$", message = "Only one or more letters are allowed!")
        String firstName,

        @Pattern(regexp = "^[a-zA-Z]+$", message = "Only one or more letters are allowed!")
        String lastName,
        @NotNull String JMBG,
        @NotNull Long dateOfBirth,
        @Pattern(regexp = "^[M|F]$", message = "Gender can be M or F!")
        String gender,
        @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
        String email,
        @Pattern(regexp = "^(\\+381|0)6\\d{7,8}$", message = "The phone number must start with +381 or 0, followed by 6 and then 7 or 8 digits!")
        String phoneNumber,
        String address,
        String password,
        @NotNull
        String position,
        @NotNull
        String department,
        List<String> roles,
        boolean active
) { }

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/dto/EmployeeCreateDTO.kt
package org.example.userservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern

data class EmployeeCreateDTO(
    val companyId: Long,
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Only one or more letters are allowed!")
    val firstName: String,
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Only one or more letters are allowed!")
    val lastName: String,
    @NotNull val JMBG: String,
    @NotNull val dateOfBirth: Long,
    @Pattern(regexp = "^[M|F]$", message = "Gender can be M or F!")
    val gender: String,
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    val email: String,
    @Pattern(
        regexp = "^(\\+381|0)6\\d{7,8}$",
        message = "The phone number must start with +381 or 0, followed by 6 and then 7 or 8 digits!"
    )
    val phoneNumber: String,
    val address: String,
    val password: String,
    @NotNull
    val position: String,
    @NotNull
    val department: String,
    val roles: List<String>,
    val active: Boolean
)
